components/button/floating-buttons.js

```diff
@@ -0,0 +1,236 @@
+				-webkit-animation: d2l-floating-buttons-animation 500ms ease-out;
+				background-color: #ffffff;
+				background-color: rgba(255, 255, 255, 0.88);
+				border-top-color: var(--d2l-color-mica);
+				bottom: 0;
+				box-shadow: 0 -2px 4px rgba(86, 90, 92, .2);
+				left: 0;
+				position: fixed;
+				right: 0;
+				z-index: 999;
+			}
+			.d2l-floating-buttons-container > div {
+				padding: 0.75rem 0 0 0;
+				position: relative;
+			}
+			.d2l-floating-buttons-container > div {
+				padding: 0.75rem 0 0 0;
+				position: relative;
+			}
+			.d2l-floating-buttons-inner-container ::slotted(d2l-button),
+			.d2l-floating-buttons-inner-container ::slotted(button),
+			.d2l-floating-buttons-inner-container ::slotted(.d2l-button) {
+				margin-right: var(--d2l-button-spacing) !important;
+				margin-bottom: var(--d2l-button-spacing) !important;
+			}
+			:host-context([dir="rtl"]) .d2l-floating-buttons-inner-container ::slotted(d2l-button),
+			:host-context([dir="rtl"]) .d2l-floating-buttons-inner-container ::slotted(button),
+			:host-context([dir="rtl"]) .d2l-floating-buttons-inner-container ::slotted(.d2l-button) {
+				margin-left: var(--d2l-button-spacing) !important;
+				margin-right: 0 !important;
+			}
+			@keyframes d2l-floating-buttons-animation {
+				0% {
+					border-color: transparent;
+					background-color: transparent;
+					transform: translate(0,10px);
+				}
+				100% {
+					border-top-color: var(--d2l-color-mica);
+					background-color: rgba(255, 255, 255, 0.88);
+					transform: translate(0,0);
+				}
+			}
+			@-webkit-keyframes d2l-floating-buttons-animation {
+				0% {
+					border-color: transparent;
+					background-color: transparent;
+					-webkit-transform: translate(0,10px);
+				}
+				100% {
+					border-top-color: var(--d2l-color-mica);
+					background-color: rgba(255, 255, 255, 0.88);
+					-webkit-transform: translate(0,0);
+				}
+			}
+			`
+		];
+	}
+
+	constructor() {
+		super();
+		this._container = null;
+		this._isRTL = false;
+		this._spacer = null;
+		this.minHeight = '500px';
+	}
+
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		window.removeEventListener('resize', this._reposition);
+		window.removeEventListener('scroll', this._reposition);
+	}
+
+	render() {
+		return html`
+			<lit-media-query
+  				.query="(max-height: ${this.minHeight})"
+  				@changed="${this._handleMediaQuery}">
+			</lit-media-query>
+			<div class="d2l-floating-buttons-container">
+				<div class="d2l-floating-buttons-inner-container">
+					<slot></slot>
+				</div>
+			</div>
+			<div class="d2l-floating-buttons-spacer"></div>
+		`;
+	}
+
+	firstUpdated() {
+		this._reposition = this._reposition.bind(this);
+		this.updateComplete.then(() => {
+
+			this._container = this.shadowRoot.querySelector('.d2l-floating-buttons-container');
+			this._spacer = this.shadowRoot.querySelector('.d2l-floating-buttons-spacer');
+
+			window.addEventListener('resize', this._reposition);
```

Comment:
These event handlers get removed in `disconnectedCallback`, so technically if the floating workflow buttons components is detached/disconnected from the DOM, and then reconnected again, these handlers would not be wired-up as needed.  I think these handlers would need to be wired up in `connectedCallback`.